MovieDB/MovieDB/Scenes/Dashboard/DashboardCoordinator.swift

```diff
@@ -0,0 +1,40 @@
+import UIKit
+
+protocol DashboardCoordinatorType {
+    func goToWatchList()
+    func goToMovieDetail(movieId: Int)
+    func goToSearch(query: String)
+}
+
+final class DashboardCoordinator: Coordinator, DashboardCoordinatorType {
+    var navigationController: UINavigationController?
+
+    init(navigationController: UINavigationController?) {
+        self.navigationController = navigationController
+    }
+
+    func start() {
+        var dashboardVC = DashboardViewController.instantiate()
+        let useCase = DashboardUseCase()
+        let viewModel = DashboardViewModel(coordinator: self,
```

Comment:
ViewModel own the Coordinator? Wrong logic. 
Coordinator's owned by ViewController.
Inside ViewController, ViewModel and Coordinator are parallel